### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by PeteUgwu**
## Set up associations

- Created classes Classroom, Book, and Rental
- Implemented the has-many/belongs-to relationship between Classroom and Student
- Implemented the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental
- Followed GitFlow

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (muneebulrehman) [inline book.rb]: `rentals` should also have a setter. For that kindly use `attr_accessor` instead of `attr_reader`.

File: person.rb [hunk 2]
```diff
@@ -1,21 +1,25 @@
 require './nameable'
 
 class Person < Nameable
+  attr_accessor :age, :name
+  attr_reader :id, :rentals
```
Comment (muneebulrehman) [inline person.rb]: `rentals` should also have a setter. For that kindly use `attr_accessor` instead of `attr_reader`.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @PeteUgwu , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR

## Required Changes ♻️
_Check the comments under the review._

- [ ] In the `Students` class, there should be a method that serves as the `belongs-to` side for the `Classroom` class, since a student belongs to a classroom. Remember to add `attr_reader` for the `classroom` variable.
image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (PeteUgwu) [reply book.rb]: Thanks for the kind feedback. Resolved

File: person.rb (hunk 2 above)
Comment (PeteUgwu) [reply person.rb]: Thanks for the kind feedback. Resolved

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @PeteUgwu, 

To highlight 👍🏽 
✔️ Associations are set correctly
✔️ Gitflow is used

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**